File: app-data/src/main/java/com/zxjdev/smile/data/moment/datasource/MomentCloudDataSource.java
package com.zxjdev.smile.data.moment.datasource;

import com.zxjdev.smile.data.moment.MomentEntity;

import java.util.List;

import javax.inject.Inject;

import rx.Observable;

public class MomentCloudDataSource {

    private MomentCloudService momentService;

    @Inject
    public MomentCloudDataSource(MomentCloudService momentService) {
        this.momentService = momentService;
    }

    public Observable<Void> addMoment(String content) {
        return momentService.addMoment(content);
    }

    public Observable<List<MomentEntity>> getMomentList() {
        return momentService.getMomentList();
    }
}

File: app-data/src/main/java/com/zxjdev/smile/data/moment/datasource/MomentCloudDataSource.kt
package com.zxjdev.smile.data.moment.datasource

import com.zxjdev.smile.data.moment.entity.MomentEntity
import javax.inject.Inject

class MomentCloudDataSource @Inject constructor(private val momentService: MomentCloudService) {

    fun getMomentList(): List<MomentEntity> {
        return momentService.getMomentList()
    }

    fun addMoment(content: String) {
        momentService.addMoment(content)
    }
}
